### DISCUSSION TITLE
Is it possible to run very big llama model?

### DISCUSSION THREAD

**Discussion by 2catycm** [ggml-org/llama.cpp#2950] [Q&A] [answered]
big means that a single GPU is not able to hold all parameters, even after we use quantimization.
In this case, parallelism and communication is needed between GPUs

**Comment by mirek190** [comment]
yes ... you can use muti gpu.

**Comment by KerfuffleV2** [comment]
You can also only offload part of the model to the GPU(s) and run the rest on CPU. Running a 70B LLaMa is possible on pure CPU with 64GB RAM.

**Comment by mirek190** [comment]
With the newest build llamacpp and  mod l 70b qk4_m  version  on rtx 3090 you can put 48 layers of 80 on GPU . I have then ~ 3 tokens /s .

**Reply by AbdullahArean** [reply]
Can you share the code how you got so fast response!

**Reply by mirek190** [reply]
I have ryzen 7950x3d , ram 7000 mhz  and rtx 3090 ... you know biffy CPU....

**Comment by staviq** [comment, marked as answer]
This already works even with multiple dissimilar GPUs, for example, I'm using rtx2070 and p106-100

In case of multiple GPUs with different amounts of VRAM, you may have to fiddle a bit with `-ts` parameter to fill the VRAM to the brim on all GPUs, but it works already.

Communication between GPUs is not strictly necessary. There is a PR #2470  for enabling "nvlink like" GPU to GPU communication, but people are reporting varying results, sometimes it's faster and sometimes slower.
